x-pack/plugins/security_solution/server/fleet_integration/handlers/remove_policy_from_trusted_apps.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ENDPOINT_TRUSTED_APPS_LIST_ID } from '@kbn/securitysolution-list-constants';
+import { ExceptionListClient } from '../../../../lists/server';
+
+interface DeletePolicy {
+  id: string;
+  name?: string;
+  success: boolean;
+}
+
+/**
+ * Removes policy from trusted apps
+ */
+export const removePolicyFromTrustedApps = async (
+  exceptionsClient: ExceptionListClient,
+  policy: DeletePolicy
+) => {
+  let page = 1;
+
+  const findTrustedAppsByPolicy = async (currentPage: number) => {
+    return exceptionsClient.findExceptionListItem({
+      listId: ENDPOINT_TRUSTED_APPS_LIST_ID,
+      filter: `exception-list-agnostic.attributes.tags:"policy:${policy.id}"`,
+      namespaceType: 'agnostic',
+      page: currentPage,
+      perPage: 50,
+      sortField: undefined,
+      sortOrder: undefined,
+    });
+  };
+
+  let findResponse = await findTrustedAppsByPolicy(page);
+  if (!findResponse) return;
+  const trustedApps = findResponse.data;
+
+  while (findResponse && (trustedApps.length < findResponse.total || findResponse.data.length)) {
+    page += 1;
+    findResponse = await findTrustedAppsByPolicy(page);
+    if (findResponse) {
```

Comment:
Isn't the paging to be reset to the previous value when no new data is fetched on a subsequent request?

Reply:
I don't think so. The `page` should always move forward here.  The `findExceptionListItem()` method will return `null` if nothing is found, so this `while` loop would be broken once/if that is the case.

Reply:
👍